packages/ui/src/app/pages/Settings/Settings.tsx

```diff
@@ -69,7 +68,85 @@ export const Settings = () => {
     }
   }, [network, endpoints])
 
-  const saveSettings = () => {
+  const checkFaucetEndpoint = async () => {
+    // check faucet endpoint
+    try {
+      const faucetStatusEndpoint = customFaucetEndpoint.replace(new RegExp('register$'), 'status')
+      const response = await fetch(faucetStatusEndpoint)
+      await response.json()
+      setIsValidFaucetEndpoint(true)
+      return true
+    } catch {
+      setIsValidFaucetEndpoint(false)
+      return false
+    }
+  }
+
+  const checkRpcEndpoint = async () => {
+    // check RPC endpoint
+    try {
+      const ws = new WebSocket(customRpcEndpoint)
+      ws.onopen = function () {
+        setIsValidRpcEndpoint(true)
+        return true
+      }
+      ws.onerror = function () {
+        setIsValidRpcEndpoint(false)
+        return false
+      }
+    } catch {
+      setIsValidRpcEndpoint(false)
+      return false
+    }
+
+    // timeout
+    await new Promise((res) => setTimeout(res, 3000))
+  }
+
+  const checkQueryEndpoint = async () => {
+    // check GraphQL endpoint
+    try {
+      const query = `
+        query {
+          electedCouncils{
+            id
+          }
+        }`
+      const response = await fetch(customQueryEndpoint, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ query }),
+      })
+      await response.json()
```

Comment:
There is a simpler way to check if it's a gql endpoint: https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/#graphql-level-health-checks

So here it would be something like:
```suggestion
      const response = await fetch(customQueryEndpoint + '?query=%7B__typename%7D')
      const succeeded = response.status < 400 && JSON.parse(await response.json()).data['__typename'] === 'Query'
```
This should be faster and also lighter on the QN